juniper_official/interface/check-ifl-state.rule

```diff
@@ -0,0 +1,144 @@
+ *      ethernet interfaces. 
+ *
+ *   2) ifl-no, is a regular expression that matches the IFL index number that
+ *      you would like to monitor.  By default it's '.*', which matches all
+ *      interfaces. Use something like '0-10*' to match only 0 to 10 IFLs.
+ *
+ */
+healthbot {
+    topic interfaces {
+        rule check-ifl-state {
+            synopsis "Sub Interface input traffic analyzer.";
+            description "Collects input/output traffic (in-octets/out-octets) periodically and notifies in case of traffic is above threshold";
+            /*
+             * Monitors the interface state. If ifl operational state is down 
+             * red alarm is shown else it is green. 
+             *
+             * Use interface and sub-interface name as keys for rule.
+             */
+            keys [interface-name sub-interface-index];
+            /*
+             * Sensor configuration to get data from network devices.
+             */
+            sensor interfaces-oc {
+                synopsis "Sub Interface open-config sensor definition";
+                description "Sub Interfaces open-config sensor to collect telemetry data from network device";
+                open-config {
+                    sensor-name /interfaces/interface/subinterfaces/subinterface/state;
+                    frequency 60s;
+                }
+            }
+            /*
+             * Fields defined using sensor path. Map the longer sensor names
+             * to the shorter field names used in the rules.
+             */
+			field interface-name {
+                sensor interfaces-oc {
+                    where "/interfaces/interface/@name =~ /{{ifd-name}}/";
+                    path "/interfaces/interface/@name";
+                }
+                type string;
+                description "Interfaces to be monitored";
+            }
+            field sub-interface-index {
+                sensor interfaces-oc {
+                    where "/interfaces/interface/subinterfaces/subinterface/@index =~ /{{ifl-no}}/";
+                    path "/interfaces/interface/subinterfaces/subinterface/@index";
+                }
+                type string;
+                description "Interfaces to be monitored";
+            }
+            field ifl-admin-status {
+                sensor interfaces-oc {
+                    path /interfaces/interface/subinterfaces/subinterface/state/admin-status;
+                }
+                type string;
+                description "Subinterface admin status";
+            }
+            field ifl-oper-status {
+                sensor interfaces-oc {
+                    path /interfaces/interface/subinterfaces/subinterface/state/oper-status;
+                }
+                type string;
+                description "Subinterface operational status";
+            }		      
+            /*
+             * Anomaly detection logic.
+             */
+            trigger ifl-state {
```

Comment:
This trigger name should be changed to more appropriate name... Something like "sub-interface-link-state"

Reply:
Changes